src/profile-logic/marker-schema.js

```diff
@@ -80,7 +80,13 @@ export function getMarkerSchemaName(
     if (type === 'tracing' && data.category) {
       // TODO - Tracing markers have a duplicate "category" field.
       // See issue #2749
-      return data.category;
+
+      // Does a marker schema for the "category" exist?
+      return markerSchemaByName[data.category]
+        ? // If so, use the category as the schema name.
+          data.category
+        : // Otherwise default back to tracing
+          'tracing';
```

Comment:
nit: This is very similar to what we do for Text markers below. I'd suggest to write the ternary in the same order. (I don't have an opinion whether this should be the `undefined` case at first or second).

optional nit: make this slightly more generic, so that if we add a new exception we don't have the error for the 3rd time. That would be something like:
```js
if (data) {
  let schemaName;

  if (type === 'tracing' && data.category) {
    schemaName = data.category
  }

  if (type === 'Text') {
    schemaName = name;
  }

  if (!schemaName || !markerSchemaByName[schemaName]) {
    schemaName = data.type;
  }

  // Coming below is me thinking out loud, this would be for other PRs:
  // newly added, would that be useful?
  // Maybe this could come first, before even `if (data)`, so that we can
  // override the type for a schema and this would replace the special
  // case for 'Text'
  if (!markerSchemaByName[schemaName]) {
    schemaName = name;
  }

  // Should we have something when we find nothing in markerSchemaByName?
  // A console.error maybe? Or a default schema to add the marker to the table
  // and the chart at least?
}
```
As you notice, I embedded some more deeper thoughts in this code ^ :-) In short it's strange to me that if we don't find a schema we just ignore the marker without any more notice (except our few special markers like GC or Network).

Reply:
I agree with the ternary order.

I'm not sure about the rest of the feedback. This function is just for finding the key to look up the schema with. The tracing and text cases are only there is migration paths to handle our different cases.